framework/src/media/audio/resample/samplerate.c

```diff
@@ -493,8 +482,15 @@ int src_simple(src_handle_t handle, src_data_t *src_data)
 			   FRAMES_TO_BYTES(src->left_frames));
 		frames_num += src->left_frames;
 
+		//according to the outsize_in_frames, to calculate the adjust_size_in_frames
+		int ratio = new_sample_rate / old_sample_rate;
+		if ((new_sample_rate % old_sample_rate) != 0) {
+			ratio += 1;
+		}
+		int adjust_size_in_frames = MINIMUM(src->size_in_frames, src->outsize_in_frames/ratio);
```

Comment:
About the resample, it is a little complex. My explaination might be a little complex too.
1. when do resample, it must  havesome remaining data, because of the filter mechanism. Need to add these remainning data to the begining of next data_in. so actual required memory size might be bigger than 'src->size_in_frames' . It will affect the next 'src->outsize_in_frames' . It is difficuclt for user to calculate the exact 'src->outsize_in_frames' which we need.
2. Because of the pcm length is not fixed and TizenRT have memory limitation, we add a split mechanism. If size_in_frames is  bigger or 'src->outsize_in_frames'  is  smaller, user can call src_simple serveral times to complete the resample in their while loop.  In the while loop ,it use src->input_frames_used related condition.
3. When need to do resample, it have a src_init(int size).  We use this size to malloc memory and calculate how many frames can resampled.   It is not equal to  'src->size_in_frames' .  If  'src->size_in_frames' is bigger,  will run into the split mechanism.
Now, it is requested to modify the interface, and user malloc data_out memory, so we modify the code to make sure resample can run normally, even if the 'src->outsize_in_frames' is smaller than required.

So,   we can not use 'ratio * src->size_in_bytes / (BYTES_PER_SAMPLE(new_sample_width) * channels_num)'  to check the 'src->outsize_in_frames' .

Reply:
Okay. I'll approve and Taejun kwon may finalize the review of this PR and and merge.